rotkehlchen/chain/ethereum/modules/thegraph/accountant.py

```diff
@@ -0,0 +1,77 @@
+from collections import defaultdict
+from collections.abc import Iterator
+from typing import TYPE_CHECKING
+
+from rotkehlchen.accounting.mixins.event import AccountingEventType
+from rotkehlchen.accounting.structures.base import get_event_type_identifier
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.ethereum.modules.thegraph.constants import CPT_THEGRAPH
+from rotkehlchen.chain.evm.accounting.interfaces import ModuleAccountantInterface
+from rotkehlchen.chain.evm.accounting.structures import TxEventSettings
+from rotkehlchen.constants import ZERO
+from rotkehlchen.fval import FVal
+
+if TYPE_CHECKING:
+    from rotkehlchen.accounting.pot import AccountingPot
+    from rotkehlchen.accounting.structures.evm_event import EvmEvent
+    from rotkehlchen.assets.asset import Asset
+    from rotkehlchen.types import ChecksumEvmAddress
+
+
+class ThegraphAccountant(ModuleAccountantInterface):
+    def reset(self) -> None:
+        self.assets_supplied: dict[ChecksumEvmAddress, FVal] = defaultdict(FVal)
+
+    def _process_deposit(
+            self,
+            pot: 'AccountingPot',  # pylint: disable=unused-argument
+            event: 'EvmEvent',
+            other_events: Iterator['EvmEvent'],  # pylint: disable=unused-argument
+    ) -> None:
+        self.assets_supplied[event.location_label] += event.balance.amount  # type: ignore[arg-type]
+
+    def _process_withdraw(
+            self,
+            pot: 'AccountingPot',
+            event: 'EvmEvent',
+            other_events: Iterator['EvmEvent'],  # pylint: disable=unused-argument
+    ) -> None:
+        self.assets_supplied[event.location_label] -= event.balance.amount
```

Comment:
This will need lots of type ignores over the file so do a cast here and use `address below`

```suggestion
        address = cast('ChecksumEvmAddress', event.location_label)  # has to be checksum address here
        self.assets_supplied[address] -= event.balance.amount
```

Reply:
Would `string_to_evm_address()` be a more preferable form for such casting in that case? Just to avoid hardcoding the target type as a string (although not much difference besides that I guess).

Reply:
You don't hard code type as a string. The reason this is there as a string and not as a type is that it's imported only for type checking.

Both solutions are ugly.

`cast()` is the lightest thing that exists by the system only for typechecking and default to no-op in realtime. But generally we also avoid that. Here it's a bit of an ugly situation due to location label being potential none for some event types.

Reply:
Ok, thanks!